tests/functional/RouteTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Tests\Functional;
+
+use PHPUnit\Framework\TestCase;
+use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerInterface;
+
+class RouteTest extends TestCase
+{
+    private static ?TestApplicationRunner $application = null;
+
+    public function testRoute()
+    {
+        self::$application = null;
+
+        $method = 'GET';
+        $url = '/';
+
+        $response = $this->doRequest($method, $url);
+
+        $this->assertInstanceOf(ResponseInterface::class, $response);
+
+        $stream = $response->getBody();
+        $stream->rewind();
+        $content = $stream->getContents();
+
+        $this->assertSame(
+            '{"status":"success","error_message":"","error_code":null,"data":{"version":"3.0","author":"yiisoft"}}',
+            $content
+        );
+    }
+
+    public function testContainer()
+    {
+        self::$application = null;
+        $container = $this->getContainer();
+
+        $this->assertInstanceOf(ContainerInterface::class, $container);
+        $this->assertTrue($container->has(LoggerInterface::class));
+    }
+
+    private function doRequest(string $method, string $url): ?ResponseInterface
```

Comment:
Do you mean moving these to a trait or to a base test case as a protected method?